synapse/lib/types.py

```diff
@@ -960,7 +963,7 @@ def _normPyStr(self, valu):
         return norm, {}
 
     def indx(self, norm):
-        return norm.encode('utf8')
+        return norm.encode('utf8', 'surrogatepass')
```

Comment:
As above, do we need to just do this everywhere? Should we make an encoder class or something?